packages/analytics/src/integrations/gtag/GTag.ts

```diff
@@ -0,0 +1,44 @@
+import { IIntegration } from '../types';
+import { IGTagConfig } from './types';
+import { loadGTag } from 'src/integrations/gtag/loadGTag';
+import { Identify, Track, Page, Group } from 'types';
+
+export class GTag implements IIntegration {
+  constructor(config: IGTagConfig) {
+    if (!config.trackingId) {
+      throw new Error('TAG ID is required for gTag');
+    }
+    loadGTag(config);
+  }
+
+  identify: Identify = (userId) => {
+    (window as any).dataLayer.push(function() {
+      this.set('userId', userId);
+    });
+    (window as any).dataLayer.push('event', 'login', { userId: userId});
+  };
+
+  // Not supported
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  group: Group = () => {};
+
+  page: Page = (pageName, traits = {}) => {
+    (window as any).dataLayer.push('event', 'page_view', {
+      page_location: traits.path || location.pathname,
+      page_title: pageName,
+    });
+  };
+
+  track: Track = (eventName, traits = {}) => {
+    (window as any).dataLayer.push('event', eventName, {
+      event_category: traits.category || 'None',
+    });
+  };
+
+  reset = () => {
+    (window as any).dataLayer.push(function() {
+      // eslint-disable-next-line @typescript-eslint/no-invalid-this
+      this.reset();
+    });
+  };
```

Comment:
is this really needed?

Reply:
yeah, it's on gtag doc: https://developers.google.com/tag-platform/tag-manager/web/datalayer#reset